driver/compile.go

```diff
@@ -14,45 +14,50 @@ import (
 	"go.uber.org/zap"
 )
 
+type Config struct {
+	Custom            proc.Compiler
+	Logger            *zap.Logger
+	ReaderSortKey     string
+	ReaderSortReverse bool
+	Span              nano.Span
+	Warnings          chan string
+}
+
 // Compile takes an AST, an input reader, and configuration parameters,
 // and compiles it into a runnable flowgraph, returning a
 // proc.MuxOutput that which brings together all of the flowgraphs
 // tails, and is ready to be Pull()'d from.
-func Compile(ctx context.Context, zctx *resolver.Context, program ast.Proc, reader zbuf.Reader, readerSortKey string, reverse bool, span nano.Span, logger *zap.Logger) (*MuxOutput, error) {
-	ch := make(chan string, 5)
-	return CompileWarningsChCustom(ctx, zctx, nil, program, reader, readerSortKey, reverse, span, logger, ch)
-}
-
-func CompileCustom(ctx context.Context, zctx *resolver.Context, custom proc.Compiler, program ast.Proc, reader zbuf.Reader, reverse bool, span nano.Span, logger *zap.Logger) (*MuxOutput, error) {
-	ch := make(chan string, 5)
-	return CompileWarningsChCustom(ctx, zctx, custom, program, reader, "", reverse, span, logger, ch)
-}
-
-func CompileWarningsCh(ctx context.Context, zctx *resolver.Context, program ast.Proc, reader zbuf.Reader, reverse bool, span nano.Span, logger *zap.Logger, ch chan string) (*MuxOutput, error) {
-	return CompileWarningsChCustom(ctx, zctx, nil, program, reader, "", reverse, span, logger, ch)
-}
+func Compile(ctx context.Context, program ast.Proc, zctx *resolver.Context, reader zbuf.Reader, cfg Config) (*MuxOutput, error) {
+	if cfg.Span.Dur == 0 {
+		cfg.Span = nano.MaxSpan
+	}
+	if cfg.Warnings == nil {
+		cfg.Warnings = make(chan string, 5)
+	}
+	if cfg.Logger == nil {
+		cfg.Logger = zap.NewNop()
+	}
```

Comment:
```suggestion
	if cfg.Logger == nil {
		cfg.Logger = zap.NewNop()
	}
	if cfg.Span.Dur == 0 {
		cfg.Span = nano.MaxSpan
	}
	if cfg.Warnings == nil {
		cfg.Warnings = make(chan string, 5)
	}
```
Nit: Follow the field declaration order.